### PR TITLE
refactor: Consolidate datafile-manager package into optimizely-sdk

### PR DESCRIPTION
**Pull request by ozayr-zaviar**
## Summary
Integrated Datafile Manager package with Optimizely SDK to remove external dependency.

## Test plan
All existing unit tests and FSC tests pass.

## Jira
[FSSDK-8205]

### CODE REVIEW

**Comment by mikechu-optimizely** [comment]
Sorry about all the code collisions @ozayr-zaviar  😬

File: packages/optimizely-sdk/karma.base.conf.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ module.exports = {
   plugins: ['karma-mocha', 'karma-webpack', require('karma-browserstack-launcher')],
 
   webpack: {
-    mode: 'production',
```
Comment (zashraf1985) [inline packages/optimizely-sdk/karma.base.conf.js:26]: can you revert it back if it works?

File: packages/optimizely-sdk/lib/plugins/datafile_manager/browser_http_polling_datafile_manager.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     if (error) {
+       logger.error(error);
+     }
+     if (configObj) {
+       datafileManagerConfig.datafile = toDatafile(configObj);
+     }
+   }
+   return new HttpPollingDatafileManager(datafileManagerConfig);
+ }
+
```
Comment (zashraf1985) [inline packages/optimizely-sdk/lib/plugins/datafile_manager/browser_http_polling_datafile_manager.ts:50]: fix end of line

File: packages/optimizely-sdk/lib/plugins/datafile_manager/http_polling_datafile_manager.tests.js [hunk 3]
```diff
@@ -16,13 +16,22 @@
...
+    // console.log(datafileManager.HttpPollingDatafileManager, {
+    //   autoUpdate: true,
+    //   datafileAccessToken: 'THE_TOKEN',
+    //   updateInterval: 5000,
+    //   urlTemplate: 'http://example.com',
+    //   sdkKey: 'SDK_KEY',
+    // })    
+    sandbox.stub(datafileManager,'HttpPollingDatafileManager')
+    // sandbox.stub(datafileManager.HttpPollingDatafileManager, 'NodeDatafileManager')
+    // sandbox.stub(datafileManager, 'NodeDatafileManager')
```
Comment (zashraf1985) [inline packages/optimizely-sdk/lib/plugins/datafile_manager/http_polling_datafile_manager.tests.js:34]: remove unneeded commented code.

File: packages/optimizely-sdk/lib/plugins/datafile_manager/react_native_http_polling_datafile_manager.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+     if (error) {
+       logger.error(error);
+     }
+     if (configObj) {
+       datafileManagerConfig.datafile = toDatafile(configObj);
+     }
+   }
+   return new HttpPollingDatafileManager(datafileManagerConfig);
+ }
+
```
Comment (zashraf1985) [inline packages/optimizely-sdk/lib/plugins/datafile_manager/react_native_http_polling_datafile_manager.ts:50]: fix end of line

**Review by zashraf1985 — CHANGES_REQUESTED** [review]
Looks great! Suggested a few minor changes. Can you update lock file version to 2.

**Review by zashraf1985 — APPROVED** [review]
Looks Great!

**Pull request merged**